engine/api/workflow_queue.go

```diff
@@ -136,15 +137,14 @@ func (api *API) postTakeWorkflowJobHandler() service.Handler {
 	}
 }
 
-func takeJob(ctx context.Context, dbFunc func() *gorp.DbMap, store cache.Store, p *sdk.Project, id int64, workerModel string, wnjri *sdk.WorkflowNodeJobRunData, wk *sdk.Worker, hatcheryName string) (*workflow.ProcessorReport, error) {
-	// Start a tx
-	tx, errBegin := dbFunc().Begin()
-	if errBegin != nil {
-		return nil, sdk.WrapError(errBegin, "Cannot start transaction")
+func takeJob(ctx context.Context, dbFunc func() *gorp.DbMap, store cache.Store, p *sdk.Project, id int64, workerModel string, wnjri *sdk.WorkflowNodeJobRunData, wk *sdk.Worker, hatcheryName string, skipProjectSecretsOnRegion []string) (*workflow.ProcessorReport, error) {
```

Comment:
to limit the number of arg of this function, we should make it a method of API struct.
this would avoid to pass dbFunc, store and this new parameter